docs/architecture/adr-038-add-support-for-wasm-light-clients.md

```diff
@@ -0,0 +1,86 @@
+# ADR 38: Add support for Wasm based light client
+
+## Changelog
+
+- 26/11/2020: Initial Draft
+
+## Status
+
+Draft
+
+## Abstract
+
+Currently in the SDK, light clients are part of Cosmos SDK codebase. What this means that, anytime a blockchain built 
+upon Cosmos SDK need to add support for new light client, Cosmos SDK need to be modified and all validator nodes of that 
+blockchain need to be updated with custom version of Cosmos SDK. 
+
+To remedy these shortcomings, we are proposing a WASM VM to host light client bytecode, which allows easier upgrading of 
+existing light clients as well as adding support for new light clients without requiring a code release and corresponding 
+hard-fork event.
+
+## Context
+Currently in the SDK, light clients are defined as part of the codebase and are implemented as submodules under 
+`/x/ibc/light-clients/`. To add support for new light client or update an existing light client in the event of security 
+issue or consensus update, we need to modify the codebase and integrate it in *numerous* places. In addition, individual 
+blockchains built upon Cosmos SDK need to update all their validator nodes to latest version to add support for this 
+light client. This entire process is tedious and time consuming. In the event a blockchain wants to add support for niche 
+light client, it need to fork Cosmos SDK and modify codebase to add support for it. This creates overhead of maintaining 
```

Comment:
If a blockchain wants to add support it just needs to register it on its codec defined in the app module

Reply:
Still requires agreement by governance and chain restart.